apps/price_pusher/src/pyth-price-listener.ts

```diff
@@ -34,108 +34,47 @@ export class PythPriceListener implements IPriceListener {
+            priceUpdate.id
+          )} ${priceUpdate.id}`
+        );
 
-    try {
-      const priceFeeds = await this.connection.getLatestPriceFeeds(
-        this.priceIds
-      );
-      priceFeeds?.forEach((priceFeed) => {
-        const latestAvailablePrice = priceFeed.getPriceUnchecked();
-        this.latestPriceInfo.set(priceFeed.id, {
-          price: latestAvailablePrice.price,
-          conf: latestAvailablePrice.conf,
-          publishTime: latestAvailablePrice.publishTime,
-        });
-      });
-    } catch (error: any) {
-      // Always log the HTTP error first
-      this.logger.error("Failed to get latest price feeds:", error);
-
-      if (error.response.data.includes("Price ids not found:")) {
-        // Extract invalid feed IDs from error message
-        const invalidFeedIds = error.response.data
-          .split("Price ids not found:")[1]
-          .split(",")
-          .map((id: string) => id.trim().replace(/^0x/, ""));
-
-        // Log invalid feeds with their aliases
-        invalidFeedIds.forEach((id: string) => {
-          this.logger.error(
-            `Price feed ${id} (${this.priceIdToAlias.get(
-              id
-            )}) not found for getLatestPriceFeeds`
-          );
-        });
+        // Consider price to be currently available if it is not older than 60s
+        const currentPrice =
+          Date.now() / 1000 - priceUpdate.price.publish_time > 60
+            ? undefined
+            : priceUpdate.price;
+        if (currentPrice === undefined) {
+          this.logger.debug("Price is older than 60s, skipping");
+          return;
+        }
 
-        // Filter out invalid feeds and retry
-        const validFeeds = this.priceIds.filter(
-          (id) => !invalidFeedIds.includes(id)
-        );
+        const priceInfo: PriceInfo = {
+          conf: currentPrice.conf,
+          price: currentPrice.price,
+          publishTime: currentPrice.publish_time,
+        };
 
-        this.priceIds = validFeeds;
+        this.latestPriceInfo.set(priceUpdate.id, priceInfo);
+        this.lastUpdated = Date.now() as TimestampInMs;
+      });
+    };
 
-        if (validFeeds.length > 0) {
-          this.logger.info(
-            "Retrying getLatestPriceFeeds with valid feeds only"
-          );
-          const validPriceFeeds = await this.connection.getLatestPriceFeeds(
-            validFeeds
-          );
-          validPriceFeeds?.forEach((priceFeed) => {
-            const latestAvailablePrice = priceFeed.getPriceUnchecked();
-            this.latestPriceInfo.set(priceFeed.id, {
-              price: latestAvailablePrice.price,
-              conf: latestAvailablePrice.conf,
-              publishTime: latestAvailablePrice.publishTime,
-            });
-          });
-        }
-      }
-    }
+    eventSource.onerror = (error: Event) => {
+      console.error("Error receiving updates from Hermes:", error);
+      eventSource.close();
+    };
```

Comment:
i think here you should either crash the service, or try to reconnect. doing otherwise will result in one dead component while the rest are working.

Reply:
It's better it crashes when the connection goes down. Just tested it.